test/common/network/address_impl_test.cc

```diff
@@ -561,10 +580,11 @@ TEST_P(MixedAddressTest, Equality) {
 }
 
 struct TestCase test_cases[] = {
-    {TestCase::Ipv4, "1.2.3.4", 1},         {TestCase::Ipv4, "1.2.3.4", 2},
-    {TestCase::Ipv4, "1.2.3.5", 1},         {TestCase::Ipv6, "01:023::00ef", 1},
-    {TestCase::Ipv6, "01:023::00ef", 2},    {TestCase::Ipv6, "01:023::00ed", 1},
-    {TestCase::Pipe, "/path/to/pipe/1", 0}, {TestCase::Pipe, "/path/to/pipe/2", 0}};
+    {TestCase::Ipv4, "1.2.3.4", 1},          {TestCase::Ipv4, "1.2.3.4", 2},
+    {TestCase::Ipv4, "1.2.3.5", 1},          {TestCase::Ipv6, "01:023::00ef", 1},
+    {TestCase::Ipv6, "01:023::00ef", 2},     {TestCase::Ipv6, "01:023::00ed", 1},
+    {TestCase::Pipe, "/path/to/pipe/1", 0},  {TestCase::Pipe, "/path/to/pipe/2", 0},
+    {TestCase::Internal, "listener_foo", 0}, {TestCase::Internal, "listener_bar", 0}};
```

Comment:
I wonder if we should have the string representation be envoy:// in config.
@htuch any thoughts here?

I could imagine for instance someone naming their listener after the address it is listening on, and there otherwise being ambiguity.

Reply:
I think for internal use in places like tests, it makes sense to have `envoy://` in the same way we have `unix://`. I think the use is limited to tests though, since in config we use the more explicit proto structure. In this specific test, I think the first element of the tuple provides the qualification.

Reply:
Good point. I will add "envoy://“ parser in test helper in the follow up